### ISSUE TITLE
Understanding roster implementation

### ISSUE THREAD

**Issue by ivucica** [UWPX/UWPX-Client#148]
## Bevor you create a new issue:

```
- [x] I searched for similar issues and did not find one
- [x] I'm using the latest version available in the [Windows Store](https://www.microsoft.com/store/apps/9NW16X9JB5WV)
```

## I'm submitting a...:

  
- Feature request (UserVoice request: )

## Current behavior:

I was looking at the source code a few months ago, trying to understand where to best integrate the address book behaviors, to integrate with My People etc. ([Reference](https://docs.microsoft.com/en-us/windows/uwp/contacts-and-calendar/integrating-with-contacts))

To my understanding, the place to do this seemed to be in `UWPX_UI` in `ConnectionHandler.C_NewRoosterMessage()`, as this is where a `RosterResultMessage` is handled in response to the relevant `<iq/>`.

However, this is only so because that function doesn't seem to maintain a roster per-se: instead of maintaining a "contact list" which gets items added and removed to (making it easy to plug in behaviors interacting with `Windows.ApplicationModel.Contacts.ContactStore`, and cleanly separating these async behaviors) -- this function maintains a list of `ChatTable`s.

## Expected behavior:

Would it make sense to maintain contacts using versioned rosters, writes to which would trigger other behaviors: updates to `ChatTable`s, creation of entries in `ContactStore` etc?

Note: Making this change to an unknown codebase written in language I'm only passingly familiar with is likely beyond the time I can reasonably dedicate to this.

## Environment:

```
App Version(s): 
4278eaf1f3ba6176940f0a1f504d6d0e62a1e295 from Sep 11 2020

Windows 10 Version Number: <!-- https://en.wikipedia.org/wiki/Windows_10_version_history -->
- [x] 2004  *note: this one is not on the list*
- [ ] 1809
- [ ] 1803
- [ ] 1709
- [ ] 1703
- [ ] 1607
- [ ] 1511
- [ ] 1507
- [ ] Insider Build (build number: )
- [ ] Misc:

Device form factor:
- [x] Desktop
- [ ] Mobile
- [ ] Xbox
- [ ] Surface Hub

Where did you got the APP from?
- [ ] [Windows Store](https://www.microsoft.com/store/apps/9NW16X9JB5WV)
- [ ] Self-build, using a provided release source
- [x] Self-build, repo cloned at [11.09.2020] <!-- When did you clone the repo! -->
- [ ] Misc, got it from... <!-- Please tell us your source! -->

```

**Comment by COM8** [comment]
It definitely makes sense to store a dedicated list of contacts somewhere.
Currently the roster gets stored in combination with with all chats.
This means each chat has a property called `inRoster` which defines if the chat is part of the users roster.
https://github.com/UWPX/UWPX-Client/blob/c494b6de13b3b9d83f70b7494ab338d0ff45de96/Storage/Classes/Models/Chat/ChatModel.cs#L43-L48

Since I'm currently reqorking the whole DB and how stuff is stored, everything related to the roster is now being handled here:
https://github.com/UWPX/UWPX-Client/blob/c494b6de13b3b9d83f70b7494ab338d0ff45de96/Manager/Classes/ClientConnectionHandler.cs#L500-L564

Here I create a change set and update the DB accordingly.
Be aware the exact way how this works will change a bit since the DB interface is not final currently. #146

So to sum up. This should be rather easy to integrate since all the magic happens in the methode above.
But I suggest you wait until #146 has be merged since here a lot of changes happen(ed).
